x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -454,6 +454,28 @@ public class Security extends Plugin
         License.OperationMode.ENTERPRISE
     );
 
+    /**
+     * 7.17.x only setting to help mitigate any potential issues for how DLS applies to terms aggs + min_doc_count=0.
+     * New versions default to stricter DLS rules and setting this to false will allow to revert to the less strict DLS behavior.
+     */
+    public static final Setting<Boolean> DLS_FORCE_TERMS_AGGS_TO_EXCLUDE_DELETED_DOCS = Setting.boolSetting(
+        "xpack.security.dls.force_terms_aggs_to_exclude_deleted_docs.enabled",
+        true,
+        Property.NodeScope,
+        Property.Deprecated
+    );
+
+    /**
+     * 7.17.x only setting to help mitigate any potential issues for how DLS applies to the validate API + rewrite=true.
```

Comment:
Maybe an odd question, but given how weird this is, is it worth putting `7x_only` right in the setting name? Just to make it really obvious if this breaks something? Just thinking out loud because man I hate the idea of retroactively introducing a breaking change to 8.x (even though I know why it's necessary here).